rpc/shared/shared/src/main/scala/io/udash/rpc/UdashRPCFramework.scala

```diff
@@ -138,56 +143,72 @@ trait UdashRPCFramework extends GetterRPCFramework with ProcedureRPCFramework wi
     }
   }
 
-  implicit val RPCResponseCodec = new GenCodec[RPCResponse] {
-    override def read(input: Input): RPCResponse = {
-      val obj = input.readObject()
-      var response: Any = null
-      var callId: String = null
-      var cause: String = null
-      var errorMsg: String = null
-      var tpe: String = null
-      while (obj.hasNext) {
-        val fi = obj.nextField()
-        fi.fieldName match {
-          case "response" =>
-            response = RawValueCodec.read(fi)
-          case "callId" =>
-            callId = fi.readString()
-          case "cause" =>
-            cause = fi.readString()
-          case "errorMsg" =>
-            errorMsg = fi.readString()
-          case "type" =>
-            tpe = fi.readString()
+  def RPCResponseCodec(exceptionsRegistry: ExceptionsRegistry): GenCodec[RPCResponse] =
```

Comment:
If this codec is not `implicit` then I think other ones should also not be.
Or they shouldn't be codecs at all, but I guess you already explained that you're planning to do this later.